topdown/cache/cache.go

```diff
@@ -67,24 +99,49 @@ type InterQueryCacheValue interface {
 // InterQueryCache defines the interface for the inter-query cache.
 type InterQueryCache interface {
 	Get(key ast.Value) (value InterQueryCacheValue, found bool)
-	Insert(key ast.Value, value InterQueryCacheValue) int
+	Insert(key ast.Value, value InterQueryCacheValue, expiresAt time.Time) int
```

Comment:
This is backwards incompatible. We could look into a new interface that is suitable for the new behavior.

Reply:
@ashutosh-narkar added a new method `InsertWithExpiry()` in the same type